testsuites/syncgateway/functional/tests/test_cbs_collections.py

```diff
@@ -328,15 +329,23 @@ def test_restricted_collection(scopes_collections_tests_fixture):
     admin_client.wait_for_db_online(db, 60)
 
     # 4. Check that documents in the server restricted collection are not accesible via SGW
-    all_docs_ids = []
-
-    for row in (sg_client.get_all_docs(sg_admin_url, db, scope=scope, collection=collection)["rows"]):
-        all_docs_ids.append(row["id"])
-    for row in (sg_client.get_all_docs(sg_admin_url, db, scope=scope, collection=second_collection)["rows"]):
-        all_docs_ids.append(row["id"])
-
-    assert(len(all_docs_ids) == 2), "Number of expected documents in Sync Gateway database does not match expected. Expected 2; Found " + str(len(all_docs_ids))
-    assert(doc_3_key not in all_docs_ids), "Sync Gateway contains document from server restricted collection. Document ID " + doc_3_key + "under " + bucket + "." + scope + "." + third_collection
+    retries = 0
+    max_retries = 15
+    while retries < max_retries:
+        all_docs_ids = []
+        for row in (sg_client.get_all_docs(sg_admin_url, db, scope=scope, collection=collection)["rows"]):
+            all_docs_ids.append(row["id"])
+        for row in (sg_client.get_all_docs(sg_admin_url, db, scope=scope, collection=second_collection)["rows"]):
+            all_docs_ids.append(row["id"])
+        try:
+            assert(len(all_docs_ids) == 2), "Number of expected documents in Sync Gateway database does not match expected. Expected 2; Found " + str(len(all_docs_ids))
+            assert(doc_3_key not in all_docs_ids), "Sync Gateway contains document from server restricted collection. Document ID " + doc_3_key + "under " + bucket + "." + scope + "." + third_collection
+            break
+        except Exception as e:
+            if retries == max_retries - 1:
+                raise e
+        sleep(1)
```

Comment:
If using a time based approach may make sense to have an exponential backoff rather than making an API request every other second

Reply:
Thanks, I agree using time is better than retry, fixed! About reporting the timeout, it slightly complicates things and the error message is informative, so how about leaving that bit out?